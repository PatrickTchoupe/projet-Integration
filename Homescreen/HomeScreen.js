import React from 'react';
import { StyleSheet, Text, View,TouchableOpacity } from 'react-native';



class HomeScreen extends React.Component {

  click_esp_pers = () => {this.props.navigation.navigate('MainChat') }
  click_cat_O = () => {this.props.navigation.navigate('catalogueOffre') }
  render(){
    return(
  <View style={styles.container}>
    <TouchableOpacity style={styles.box1}>
      <Text style={styles.text}>Demande de Service</Text>
    </TouchableOpacity>
  
      <TouchableOpacity style={styles.box2} onPress = {this.click_cat_O}>
        <Text style={styles.text}>Offre de Service</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.box3} onPress = {this.click_esp_pers} >
        <Text style={styles.text}>Espace personnel</Text>
      </TouchableOpacity>
    
  </View>
  )
}
}

const styles = StyleSheet.create({
  container: {
      width : "100%",
      height : "100%",
    backgroundColor: 'rgba(106,106,255,0.6)',
  },
  box1: {
    position: 'absolute',
    top: 250,
    left: 40,
    width: 150,
    height: 150,
    backgroundColor: 'rgba(242,100,107,1)',
    borderRadius : 20,
  },
  box2: {
    ...StyleSheet.absoluteFill,
    width: 150,
    height: 150,
    top: 250,
    left: 200,
    backgroundColor: 'rgba(94,94,205,1)',

    borderRadius : 20,
  },
  box3: {
    position: 'absolute',
    top: 420,
    left: 40,
    width: 310,
    height: 150,
    backgroundColor: 'rgba(0,128,0,0.5)',

    borderRadius : 20,
  },
  text: {
    color: '#FFF',
    fontSize: 25,
    textAlign: "center",
    marginTop: 40,
  }
});

export default HomeScreen;