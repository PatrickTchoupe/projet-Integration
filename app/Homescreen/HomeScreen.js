/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { Button, Overlay } from 'react-native-elements';
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native';
import MaterialCommunityIconsIcon from 'react-native-vector-icons/MaterialCommunityIcons';



function Mainscreen(props) {
  const handlePress = () => {
    props.navigation.navigate('Discussion_Repo');
  };
  const handleProfil = () => {
    props.navigation.navigate('Profil');
  };

  const handleDemande = () => {
    props.navigation.navigate('Demande');
  };

  const handleListeDem = () => {
    props.navigation.navigate('ListeDem');
  };
    
  return (

    <View style={styles.container}>
      <View style={styles.buttonRow}>
        <TouchableOpacity style={styles.button} onPress={handleListeDem}>
          <Text style={styles.joffreMonService1}>
            J&#39;offre mon{'\n'}service
          </Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button2} onPress={handleDemande}>
          <Text style={styles.demandeDaide}>Demande {'\n'}d&#39;aide</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.button3Row}>
        <TouchableOpacity style={styles.button3} onPress={handleProfil}>
          <Text style={styles.profile}>Profile</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={handlePress}
          style={styles.button4}
        >
          <Text style={styles.discussion}>Discussion</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.iconStack}>
        <MaterialCommunityIconsIcon
          name='home'
          style={styles.icon}
        />
        <MaterialCommunityIconsIcon
          name="home"
          style={styles.icon3}
        />
        <MaterialCommunityIconsIcon
          name="account-circle"
          style={styles.icon4}
        />
      </View>
      <MaterialCommunityIconsIcon
        name="home"
        style={styles.icon2}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  button: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(227,144,153,1)',
    borderRadius: 22
  },
  joffreMonService1: {
    fontFamily: 'roboto-300',
    color: '#121212',
    lineHeight: 18,
    textAlign: 'center',
    fontSize: 18,
    marginTop: 77,
    marginLeft: 40
  },
  button2: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(118,171,227,1)',
    borderRadius: 22,
    marginLeft: 23,
    marginTop: 2
  },
  demandeDaide: {
    fontFamily: 'roboto-300',
    color: '#121212',
    fontSize: 18,
    textAlign: 'center',
    marginTop: 75,
    marginLeft: 41
  },
  buttonRow: {
    height: 196,
    flexDirection: 'row',
    marginTop: 157,
    marginLeft: 9,
    marginRight: 4
  },
  button3: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(189,244,123,1)',
    borderRadius: 22
  },
  profile: {
    fontFamily: 'arial-regular',
    color: '#121212',
    fontSize: 18,
    marginTop: 73,
    marginLeft: 58
  },
  button4: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(183,154,209,1)',
    borderRadius: 22,
    marginLeft: 23
  },
  discussion: {
    fontFamily: 'roboto-300',
    color: '#121212',
    fontSize: 21,
    textAlign: 'center',
    marginTop: 73,
    marginLeft: 30
  },
  button3Row: {
    height: 194,
    flexDirection: 'row',
    marginTop: 10,
    marginLeft: 9,
    marginRight: 4
  },
  icon: {
    top: 0,
    left: 38,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40
  },
  icon3: {
    top: 36,
    left: 33,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40
  },
  icon4: {
    top: 42,
    left: 0,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40
  },
  iconStack: {
    width: 78,
    height: 85,
    marginTop: -576,
    marginLeft: -105
  },
  icon2: {
    color: 'rgba(128,128,128,1)',
    fontSize: 40,
    marginTop: -143,
    marginLeft: 344
  }
});

export default Mainscreen;
