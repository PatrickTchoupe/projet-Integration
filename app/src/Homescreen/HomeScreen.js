/* eslint-disable prettier/prettier */
import React, {useEffect} from 'react';
import {StyleSheet, View, TouchableOpacity, Text} from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import LinearGradient from 'react-native-linear-gradient';

//import MaterialCommunityIconsIcon from 'react-native-vector-icons/MaterialCommunityIcons';
let userId = ' ';
function Homescreen(props) {
  async function getContact() {
    let id = await AsyncStorage.getItem('user');
    id = JSON.parse(id).Id;
    userId = id;
    fetch(`https://help-recover-api.herokuapp.com/users/${userId}`)
      .then((reponse) => reponse.json())
      .then((json) => {
        if (json[0].password === '85550') {
          props.navigation.navigate('Connexion');
        }
      });
  }
  useEffect(() => {
    getContact();
  });
  const handlePress = () => {
    props.navigation.navigate('Discussion_Repo', {screen: 'disc'});
  };
  const handleProfil = () => {
    props.navigation.navigate('Profil');
  };

  const handleDemande = () => {
    props.navigation.navigate('Demande');
  };

  const handleListeDem = () => {
    props.navigation.navigate('ListeDem');
  };

  return (
    <LinearGradient
      start={{x: 0, y: 0.1}}
      end={{x: 0.9, y: 0.9}}
       locations={[0, 0.4, 0.8]}
      colors={['#bde0fe','#ffffff', '#bde0fe']}
      style={styles.linearGradient}>
      <View style={styles.buttonRow}>
        <TouchableOpacity style={styles.button} onPress={handleListeDem}>
          <Text style={styles.joffreMonService1}>J&#39;offre mon{'\n'}service</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button2} onPress={handleDemande}>
          <Text style={styles.demandeDaide}>Demande de Service</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.button3Row}>
        <TouchableOpacity style={styles.button3} onPress={handleProfil}>
          <Text style={styles.profile}>Profil</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={handlePress} style={styles.button4}>
          <Text style={styles.discussion}>Discussion</Text>
        </TouchableOpacity>
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  linearGradient: {
    flex: 1,
    paddingLeft: 15,
    paddingRight: 15,
    borderRadius: 5,
  },
  button: {
    width: '46%',
    height: 194,
    backgroundColor: '#7CDB8A',

    // backgroundColor: 'rgba(227,144,153,1)',
    borderRadius: 22,
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
  joffreMonService1: {
    fontFamily: 'roboto-300',
    color: '#121212',
    lineHeight: 18,
    textAlign: 'center',
    fontSize: 18,
    marginTop: 77,
    marginLeft: 20,
  },
  button2: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(118,171,227,1)',
    borderRadius: 22,
    marginLeft: 23,
    marginTop: 2,
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,

  },
  demandeDaide: {
    fontFamily: 'roboto-300',
    color: '#121212',
    fontSize: 18,
    textAlign: 'center',
    marginTop: 75,
    marginLeft: 20,
  },
  buttonRow: {
    height: 196,
    flexDirection: 'row',
    marginTop: 157,
    marginLeft: 9,
    marginRight: 4,
  },
  button3: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(189,244,123,1)',
    borderRadius: 22,
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
  profile: {
    fontFamily: 'arial-regular',
    color: '#121212',
    fontSize: 18,
    marginTop: 73,
    marginLeft: 58,
  },
  button4: {
    width: '46%',
    height: 194,
    backgroundColor: 'rgba(183,154,209,1)',
    borderRadius: 22,
    marginLeft: 23,
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
  discussion: {
    fontFamily: 'roboto-300',
    color: '#121212',
    fontSize: 21,
    textAlign: 'center',
    marginTop: 73,
    marginLeft: 20,
  },
  button3Row: {
    height: 194,
    flexDirection: 'row',
    marginTop: 10,
    marginLeft: 9,
    marginRight: 4,
  },
  icon: {
    top: 0,
    left: 38,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40,
  },
  icon3: {
    top: 36,
    left: 33,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40,
  },
  icon4: {
    top: 42,
    left: 0,
    position: 'absolute',
    color: 'rgba(128,128,128,1)',
    fontSize: 40,
  },
  iconStack: {
    width: 78,
    height: 85,
    marginTop: -576,
    marginLeft: -105,
  },
  icon2: {
    color: 'rgba(128,128,128,1)',
    fontSize: 40,
    marginTop: -143,
    marginLeft: 344,
  },
});

export default Homescreen;
