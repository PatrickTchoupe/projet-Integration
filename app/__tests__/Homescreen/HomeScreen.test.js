/* eslint-disable import/imports-first */
/* eslint-disable import/no-unresolved */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import React from 'react';
import HomeScreen from './../../HomeScreen/HomeScreen';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
