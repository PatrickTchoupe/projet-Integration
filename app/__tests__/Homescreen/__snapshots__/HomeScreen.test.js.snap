// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 196,
        "marginLeft": 9,
        "marginRight": 4,
        "marginTop": 157,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(227,144,153,1)",
          "borderRadius": 22,
          "height": 194,
          "opacity": 1,
          "width": "46%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "roboto-300",
            "fontSize": 18,
            "lineHeight": 18,
            "marginLeft": 40,
            "marginTop": 77,
            "textAlign": "center",
          }
        }
      >
        J'offre mon
        

        service
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(118,171,227,1)",
          "borderRadius": 22,
          "height": 194,
          "marginLeft": 23,
          "marginTop": 2,
          "opacity": 1,
          "width": "46%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "roboto-300",
            "fontSize": 18,
            "marginLeft": 41,
            "marginTop": 75,
            "textAlign": "center",
          }
        }
      >
        Demande 
        

        d'aide
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 194,
        "marginLeft": 9,
        "marginRight": 4,
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(189,244,123,1)",
          "borderRadius": 22,
          "height": 194,
          "opacity": 1,
          "width": "46%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial-regular",
            "fontSize": 18,
            "marginLeft": 58,
            "marginTop": 73,
          }
        }
      >
        Profile
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(183,154,209,1)",
          "borderRadius": 22,
          "height": 194,
          "marginLeft": 23,
          "opacity": 1,
          "width": "46%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "roboto-300",
            "fontSize": 21,
            "marginLeft": 30,
            "marginTop": 73,
            "textAlign": "center",
          }
        }
      >
        Discussion
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 85,
        "marginLeft": -105,
        "marginTop": -576,
        "width": 78,
      }
    }
  >
    <Text />
    <Text />
    <Text />
  </View>
  <Text />
</View>
`;
